Add generic write function and some fixes

paretje opened the pull request:
- fix some issues with new config parser
- add `write_results` function to `spectrum_output` accepting a list of output formats

paretje committed:
support config extension

paretje committed:
fix usage of config parser in main

paretje committed:
don't fail silently when config file doesn't exist

paretje committed:
add write_csv and general write_results function

RalfG commented on ms2pip/ms2pip_tools/spectrum_output.py:
```diff
@@ -61,6 +76,8 @@ class SpectrumOutput:
         Write predictions to Spectronaut CSV file
+    write_csv()
+        Write predictions to CSV file
 
```
Maybe we should also add the `write_results` method to the docstring

RalfG reviewed (changes requested):
Add new method to docstring in spectrum_output

paretje committed:
replace self with decorated function

As we don't need to change anything to the decorated function at
runtime, we just want to keep track of a dictionary in the class of the
function, we can just replace us, the decorator, with the actual
undecorated function.

Otherwise, we could have used:
```python
def output_format(output_format):
    class OutputFormat:
        def __init__(self, fn):
            self.fn = fn
            self.output_format = output_format

        def __set_name__(self, owner, name):
            owner.OUTPUT_FORMATS[self.output_format] = self.fn

        def __call__(self, *args, **kwargs):
            return self.fn(*args, **kwargs)

        def __get__(self, obj, objtype):
            '''Support instance methods.'''
            return functools.partial(self.__call__, obj)
```

paretje committed:
add documentation

RalfG committed:
Merge branch 'releases' into feature/write-results

paretje committed:
return spectronaut file object

paretje committed:
match output format lower case

RalfG commented:
Can this one be merge, @paretje?
